templates/addons/s3/policy.yml

```diff
@@ -0,0 +1,23 @@
+{{.BucketName}}AccessPolicy:
+  Type: AWS::IAM::ManagedPolicy
+  Properties:
+    Description: !Sub
+      - Grants CRUD access to the S3 bucket ${Bucket}
+      - { Bucket: !Ref {{.BucketName}} }
+    PolicyDocument:
+      Version: 2012-10-17
+      Statement:
+        - Sid: {{.BucketName}}-s3-object-actions
+          Effect: Allow
+          Action:
+          - 's3:GetObject'
+          - 's3:PutObject'
+          - 's3:PutObjectACL'
+          - 's3:PutObjectTagging'
+          - 's3:DeleteObject'
+          - 's3:RestoreObject'
+          Resource: 'arn:aws:s3:::{{.BucketName}}/*'
+        - Sid: {{.BucketName}}-s3-list-action
+          Effect: Allow
+          Action: 's3:ListBucket'
+          Resource: 'arn:aws:s3:::{{.BucketName}}'
```

Comment:
Is the S3 bucket manually created by users or we'll create one for them? I wonder if we need permissions to create/delete the s3 bucket as well 🤔

Reply:
We're creating one for them through CF. This policy is strictly the actions that the task role will need to access the objects in the bucket.